### ISSUE TITLE
Ice Jam Services http protocol error

### ISSUE THREAD

**Issue by lprivette** [USGS-WiM/IceJamsservices#7]
Could this be related to the SSL cipher fix. too?

I attempted to create a new Task which has worked before in resolving service issues: https://github.com/USGS-WiM/IceJamsservices/wiki/Restart-EC2-FARGATE-instance ( I have since stopped the new task I had created and left the old ones running

Request in network tab are going through technically https://icejams.wim.usgs.gov/iceconditiontypes
Client is reporting this error in browser: 
![image](https://user-images.githubusercontent.com/23640741/162041927-62500ede-dc68-4cf6-9323-c4d22ae89928.png)

I am not on VPN right now.

Looking at the task in the console this looks like it's at max if I'm reading the bars right: 
![image](https://user-images.githubusercontent.com/23640741/162042162-e7679df0-793a-4ba8-b578-4dbaab4f71e8.png)

I think  the console has changed cause I can't see the cluster vpcs or the subnets in the old task to verify I'm adding the right ones to the new task.

**Comment by aaronstephenson** [comment]
I don't recall much about this project. Let's meet to discuss it, check my calendar and propose a time.

**Comment by ChadFanguy-usgs** [comment]
I don't think the ssl changes should effect this since I reverted the changes but I'm confused how this is running; is it running in FARGATE?

**Comment by lprivette** [comment]
Yes it's running in Fargate. Maybe I should just mess with the tasks again. For step 6 where I mention the subnet, I state that there should be two but even in the existing tasks there is  only one. 

is it possible a subnet was deleted?

**Comment by aaronstephenson** [comment]
I took a look at the WMA subaccount's VPCs, and those two subnets are in the "default" VPC and allow all traffic. I've never used that VPC, tho. The production EC2 instances use one of two VPCs created by Zivaro that are heavily locked down (e.g., requires bastions), and our dev/test EC2 instances use a VPC I created specifically for them that is not so locked down (can remote without a bastion). This is likely a red herring, tho.

The real problem is that the load balancer in front of this ECS task only allows traffic over port 80. Was that intentional? That's why you're getting a protocol error in your request, because you're requesting over port 443.

**Comment by lprivette** [comment]
I believe it was intentional. I remember Jeremy troubleshooting for a while to get these services working on Fargate. I'm search for a record of his process now.

I may need an ELI5 I've never specified a port in requests made by clients. So how would we ensure that we're requesting over port 80?

**Comment by aaronstephenson** [comment]
By using http instead of https. But really, no one should use http except in local development or if you can guarantee that the traffic is all back-end and behind a proxy that handles all external requests over https. The load balancer should be https, if nothing else.

**Comment by lprivette** [comment]
Ok can I meet with one of you on a call today to attempt to resolve this? I don't understand what specifically I need to do in AWS to resolve this -- sounds like the load balancer needs adjustments? I haven't made any changes to it since Jeremy started it up years ago so I'm surprised I would suddenly need to make calls over http when I've been doing them over https.

**Comment by lprivette** [comment]
I tried switching to https locally and the same error occurred. With some googling it states that the disk space being full can cause this error, so I'm messing with creating new tasks again.

This app is being demo'd on the WMA Seminar Series on the 13th. Would greatly appreciate any help getting this back up for the people demoing it when you can.

**Comment by lprivette** [comment]
i kinda fixed it by updating the service from 1 task to 5. https://test.wim.usgs.gov/icejams/#/home. I do not know if this has any effect on cost, do you all?

**Comment by lprivette** [comment]
Fixed by "Updating" the services and upping the tasks to 6

**Comment by lprivette** [comment]
@aaronstephenson need permission to view load balance stats
![image](https://user-images.githubusercontent.com/23640741/162521827-5a2b4403-adf1-40de-9681-2d636538f617.png)

**Comment by aaronstephenson** [comment]
IAM permissions updated, try viewing load balancers again @lprivette.

**Comment by lprivette** [comment]
Happened again today. I upped the desired tasks to 20 and set up autoscaling (which wasn't enabled). I'm worried this isn't acutally fixing the issue though.

**Comment by lprivette** [comment]
@ChadFanguy-usgs @aaronstephenson This is continuing to happen today so I'm not sure the autoscaling is helping. This is just a prototype app but it's getting embarrassing at this point with people trying to actively test/use it.

Should I try to deploy the services outside of fargate or something?

**Comment by aaronstephenson** [comment]
I'm sorry this keeps happening @lprivette. I wasn't involved in setting this up, and I don't have any real experience with these AWS services, so I can't do much else to help unless I get approval to spend significant hours learning them. Perhaps for now, we just try to deploy your app in a traditional manner, via S3 or EC2?

**Comment by lprivette** [comment]
Any chance we'd be using Fargate/ECS in any other or future projects? I'm wondering if it's worth it to put the time into actually understanding what's going on, but if we're not planning on using it going forward then we should def move it to an s3 instance.

**Comment by aaronstephenson** [comment]
ECS and Fargate are definitely services I want to explore more, but I just don't have time available at the moment to learn more about them. I was only recommending moving this app to S3 or EC2 as a stopgap measure until we can spend the time to figure this out.
